runtime/lua/vim/lsp/buf.lua

```diff
@@ -213,10 +213,15 @@ function M.format(options)
         return
       end
       local params = util.make_formatting_params(options.formatting_options)
-      client.request('textDocument/formatting', params, function(...)
+      client.request('textDocument/formatting', params, function(err, result, ctx, config)
         local handler = client.handlers['textDocument/formatting']
           or vim.lsp.handlers['textDocument/formatting']
-        handler(...)
+        local mode = vim.api.nvim_get_mode().mode
+        if changedtick ~= vim.api.nvim_buf_get_var(ctx.bufnr, 'changedtick') or mode:find('^i') then
+          return
```

Comment:
What do you think of adding a vim.notify to have some sort of feedback about this? Otherwise users may wonder why nothing happened

Reply:
Good point, I'll add this.
The async path actually has no `vim.notify` on error at all. I'll also add this to the handler.